From 94f923fffdc7d6f682f0c1accdb0aef0ada00e6d Mon Sep 17 00:00:00 2001
From: polirritmico <ebray187@gmail.com>
Date: Wed, 28 Sep 2022 10:17:57 -0300
Subject: [PATCH] effects/desktopgrid: Change mouse click behaviour

Change the mouse click behaviour of right click to close the window and
middle click to pin the window across desktops.
---
 src/effects/desktopgrid/qml/DesktopView.qml | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/effects/desktopgrid/qml/DesktopView.qml b/src/effects/desktopgrid/qml/DesktopView.qml
index 771dcc8..d44d36c 100644
--- a/src/effects/desktopgrid/qml/DesktopView.qml
+++ b/src/effects/desktopgrid/qml/DesktopView.qml
@@ -121,13 +121,14 @@ FocusScope {
         }
         onActivated: effect.deactivate(effect.animationDuration);
         onWindowClicked: {
-            if (eventPoint.event.button !== Qt.MiddleButton) {
-                return;
-            }
-            if (window.desktop > -1) {
-                window.desktop = -1;
-            } else {
-                window.desktop = desktopView.desktop.x11DesktopNumber;
+            if (eventPoint.event.button === Qt.MiddleButton) {
+                if (window.desktop > -1) {
+                    window.desktop = -1;
+                } else {
+                    window.desktop = desktopView.desktop.x11DesktopNumber;
+                }
+            } else if (eventPoint.event.button === Qt.RightButton) {
+                window.closeWindow();
             }
         }
         Behavior on x {
-- 
2.35.1

